code/game/machinery/pipe/construction.dm

```diff
@@ -159,33 +159,26 @@ Buildable meters
 	var/flags = initial(fakeA.pipe_flags)
 	var/pipe_count = 0
 	for(var/obj/machinery/atmospherics/machine in loc)
-		if(machine.piping_layer != piping_layer)
+		if(machine.piping_layer != piping_layer && !((machine.pipe_flags | flags) & PIPING_ALL_LAYER))
 			continue
 		pipe_count += 1
 	for(var/obj/machinery/atmospherics/machine in loc)
 		if((machine.pipe_flags & flags & PIPING_ONE_PER_TURF)) //Only one dense/requires density object per tile, eg connectors/cryo/heater/coolers.
 			to_chat(user, span_warning("Something is hogging the tile!"))
 			return TRUE
 
-		if(pipe_count == 1 && istype(machine, /obj/machinery/atmospherics/pipe/smart) && ispath(pipe_type, /obj/machinery/atmospherics/pipe/smart) && lowertext(machine.pipe_color) != lowertext(pipe_color) && machine.connection_num < 3)
-			var/direction = machine.dir
-			if((direction & EAST|WEST || direction & SOUTH|NORTH) && !ISDIAGONALDIR(direction))
-				pipe_type = /obj/machinery/atmospherics/pipe/bridge_pipe
-				if(EWCOMPONENT(direction))
-					dir = NORTH
-				if(NSCOMPONENT(direction))
-					dir = EAST
-				continue
-
-		if(flags & PIPING_BRIDGE && !(machine.pipe_flags & PIPING_BRIDGE) && check_ninety_degree_dir(machine)) //continue if we are placing a bridge pipe over a normal pipe only (prevent duplicates)
+		// skip checks if we don't overlap layers, either by being on the same layer or by something being on all layers
+		if(machine.piping_layer != piping_layer && !((machine.pipe_flags | flags) & PIPING_ALL_LAYER))
```

Comment:
I don't like that you need to dupe this check. Could you make pipe_count into like a list of acceptable machines and then iterate over that, using its length where appropriate?

Reply:
Done